### PR TITLE
Make tgui alerts no longer automatically focus

### LINKED ISSUE #59517: Pop-ups steal focus, leading to many accidental ghost roles
## Reproduction:
Be in the middle of typing something as an observer. 
Have a ghost role prompt pop up. 
Hit space because you're in the middle of typing. 
Accidentally hit "yes" and get the ghost role you didn't want. 
Suffer.

### PR DESCRIPTION
**Pull request by RandomGamer123**
## About The Pull Request
TGUI alerts no longer automatically focus. 
An autofocus parameter was also added to the AlertModal class which can be set to restore the old behavior of automatically focusing.
TGUI alerts still focus (and highlight the option) when clicked on.

## Why It's Good For The Game
Fixes #59517
Allows for more flexibility with TGUI alerts while preventing people accidentally clicking them (most notably with ghost role spawners) by pressing space (such as when typing).

## Changelog
:cl:
qol: TGUI pop-up alerts no longer automatically focus by default until you click on them. This prevents people from accidentally clicking an option they did not intend to pick by pressing space.
code: TGUI pop-ups now have a new parameter to focus automatically by default.
/:cl:

### CODE REVIEW

**Comment by brotherbeyondd** [comment]
I love you

File: tgui/packages/tgui/interfaces/AlertModal.js [hunk 1]
```diff
@@ -18,11 +18,12 @@ import {
 } from 'common/keycodes';
 
 export class AlertModal extends Component {
-  constructor() {
+  constructor(autofocus = false) {
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/AlertModal.js:21]: This does not work the way you think it does, you should remove this from here, and perhaps just make it a constant.

**Comment by LemonInTheDark** [comment]
I wonder if it would be better to make the default focusing rather then unfocusing, since that way any issues will become apparent quickly

**Comment by RandomGamer123** [comment]
> I wonder if it would be better to make the default focusing rather then unfocusing, since that way any issues will become apparent quickly

Well I would say in most cases you wouldn't want the button to autofocus since the risk of mispressing it is quite high, especially as this modal is used for alerts

**Comment by LemonInTheDark** [comment]
I guess the stuff you want to autofocus is popups prompted by the user, and the stuff you want to not is stuff prompted out of their control. I'll confess I'm not sure what that list is tho

**Comment by JJRcop** [comment]
Doesn't `alert` autofocus? It should keep that behavior by default.

I say just unfocus ghost role notifications, but leave everything else. This is a negative QoL change for all of the admin prompts.

**Comment by JJRcop** [comment]
This PR does not expose any DM interface to control autofocus.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
> I say just unfocus ghost role notifications, but leave everything else.

I agree with this. 

I pressed enter on the observe prompt, and would like to keep that.

**Comment by JJRcop** [comment]
There must be a DM-side interface to this, as you otherwise cannot change the JS value of `autofocus`.

**Comment by Mothblocks** [comment]
:pensive:

**Comment by LemonInTheDark** [comment]
😔

**Pull request closed without merging**